src/main/java/net/dries007/tfc/util/loot/CustomItemPredicate.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.util.loot;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import net.minecraft.advancements.critereon.*;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TieredItem;
+
+import net.minecraftforge.fluids.FluidStack;
+
+import net.dries007.tfc.TerraFirmaCraft;
+import net.dries007.tfc.common.recipes.HeatingRecipe;
+import net.dries007.tfc.common.recipes.inventory.ItemStackInventory;
+import net.dries007.tfc.util.Helpers;
+import net.dries007.tfc.util.JsonHelpers;
+import net.dries007.tfc.util.Metal;
+import org.jetbrains.annotations.Nullable;
+
+public class CustomItemPredicate extends ItemPredicate
```

Comment:
I want to say this entire class is unnecessary. It is hardcoding what should really just be a tag, and it doesn't even look like it's an efficient or better version of just specifying a tag here.